pkg/keygen/keygen.go

```diff
@@ -66,12 +73,49 @@ func (k *Generator) KeyForMetadata(meta metadata.Metadata) (crypto.PrivateKey, e
 	pk, err := pki.DecodePrivateKeyBytes(bytes)
 	if err != nil {
 		// Generate a new key if the existing one cannot be decoded
-		return new2048BitRSAKey()
+		return newKey(attrs)
 	}
 
 	return pk, nil
 }
 
-func new2048BitRSAKey() (crypto.PrivateKey, error) {
-	return rsa.GenerateKey(rand.Reader, 2048)
+func newKey(attrs map[string]string) (crypto.PrivateKey, error) {
+	switch algo := attrs[csiapi.KeyAlgorithmKey]; algo {
+	case string(cmapi.RSAKeyAlgorithm):
```

Comment:
suggestion: We might need to also handle the algorithm casing here, using ToUpper. Same deal with Ed25519 needing ToUpper as well.